ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -152,17 +152,18 @@ public void writeRecord(final ConsumerRecord<byte[], byte[]> record) {
 
   void writeCommandToBackup(final ConsumerRecord<CommandId, Command> record) {
     if (corruptionDetected) {
-      LOG.warn("Failure to write command topic data to backup. "
-          + "Corruption detected in command topic.");
-      return;
+      throw new KsqlServerException(
+          "Failed to write record due to out of sync command topic and backup file: " + record);
```

Comment:
Would be good to use the same message here as the one in the LOG.warn().

Reply:
done